### PR TITLE
Feat/321 skip overlay segmented graph

### PR DESCRIPTION
**Pull request by ArdtK**
## Issue addressed
Solves #321

## Code of conduct
- [x] I HAVE NOT added sensitive or compromised (test) data to the repository.
- [x] I HAVE NOT added vulnerabilities to the repository.
- [x] I HAVE discussed my solution with (other) members of the RA2CE team.

## What has been done?
1. Based on the new setting `NetworkConfigData.Hazard.overlay_segmented_network` (bool) the hazard overlay of the base (segmented) network is done. Default value is `True` (hazard overlay is created).
2. If the hazard overlay on the base (segmented) network is created, the files should be exported (`base_network_hazard.*`).

### Checklist
- [x] Code is formatted using our custom `black` and `isort` definitions.
- [x] Tests are either added or updated.
- [x] Branch is up to date with `master`.
- [x] Updated documentation if needed.

## Additional Notes (optional)
* split HazardOverlay.create() into smaller methods
* add locations_hazard to GraphFileCollection to facilitate export
* replace networkx.classes.*graph.<GraphType> by networkx.<GraphType> for readability
* small improvements to the NetworkExporter classes

### CODE REVIEW

File: ra2ce/network/hazard/hazard_overlay.py [hunk 1]
```diff
@@ -490,41 +369,139 @@ def load_origins_destinations(self):
...
+            _base_graph_hazard_reprojected = self.hazard_intersect(_graph_reprojected)
+
+            # Assign the original geometries to the reprojected raster
+            _graph_hazard = self.get_original_geoms_graph(
+                base_graph, _base_graph_hazard_reprojected
+            )
+        else:
+            _graph_hazard = self.hazard_intersect(base_graph)
+
+        return _graph_hazard
```
Comment (Carsopre) [inline ra2ce/network/hazard/hazard_overlay.py:401]:
```suggestion
            return self.get_original_geoms_graph(
                base_graph, _base_graph_hazard_reprojected
            )
        return self.hazard_intersect(base_graph)
```

File: ra2ce/network/network_config_data/network_config_data.py [hunk 2]
```diff
@@ -82,6 +82,9 @@ class HazardSection:
     aggregate_wl: AggregateWlEnum = field(default_factory=lambda: AggregateWlEnum.NONE)
     hazard_crs: str = ""
     scenario_cost: list[float] = field(default_factory=list)
+    # Should a hazard overlay be created for the segmented network.
```
Comment (Carsopre) [inline ra2ce/network/network_config_data/network_config_data.py:85]: Make it more directive `When true the overlay network is segmented`.

Comment (ArdtK) [reply ra2ce/network/network_config_data/network_config_data.py:85]: Done

File: ra2ce/network/exporters/network_exporter_factory.py [hunk 3]
```diff
@@ -38,27 +38,30 @@
...
+        _exporter_type = self.get_exporter_type(network)
         self._exporter = _exporter_type(basename, export_types)
         self._exporter.export(output_dir, network)
 
     def get_pickle_path(self) -> Path:
         return self._exporter.pickle_path
 
     @staticmethod
-    def get_exporter(network: NETWORK_TYPE) -> Type[NetworkExporterBase]:
+    def get_exporter_type(network: NETWORK_TYPE) -> Type[NetworkExporterBase]:
```
[the remark this answers is not in the document]
Comment (ArdtK) [reply ra2ce/network/exporters/network_exporter_factory.py:58]: Yes

File: ra2ce/network/hazard/hazard_overlay.py (hunk 1 above)
Comment (ArdtK) [reply ra2ce/network/hazard/hazard_overlay.py:401]: Done on several locations

**Pull request merged**